File: create_tree_from_dict.py
import os
import json

def create_tree_from_dict(tree, root):
    for key, value in tree.items():
        if key == "_files":
            for file in value:
                open(os.path.join(root, file), 'w').close()
        else:
            dir_path = os.path.join(root, key)
            if not os.path.exists(dir_path):
                os.makedirs(dir_path)
            create_tree_from_dict(value, dir_path)

def main():
    with open("filesystem.json", "r") as f:
        tree = json.load(f)
    root_path = "."  # Remplacer par le chemin du répertoire racine où recréer l'arborescence
    create_tree_from_dict(tree, root_path)

if __name__ == "__main__":
    main()

File: createTreeFromDict.js
const fs = require('fs');
const path = require('path');

function createTreeFromDict(tree, rootPath) {
    for (const key in tree) {
        if (tree[key] !== null && typeof tree[key] === 'object') {
            const newDirPath = path.join(rootPath, key);
            if (!fs.existsSync(newDirPath)) {
                fs.mkdirSync(newDirPath, { recursive: true });
            }
            createTreeFromDict(tree[key], newDirPath);
        }
    }
}

function main() {
    const jsonPath = "./filesystem.json";  // Le fichier JSON généré
    const outputPath = "./output_directory";  // Le répertoire où recréer l'arborescence

    const tree = JSON.parse(fs.readFileSync(jsonPath, 'utf8'));

    if (!fs.existsSync(outputPath)) {
        fs.mkdirSync(outputPath, { recursive: true });
    }

    createTreeFromDict(tree, outputPath);
}

main();
